ci/bootstrap-ubuntu-20.04.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+# This bootstrap script is used by both the Github CI action and the
+# Vagrant VM configuration.
+
+set -xeo pipefail
+sudo apt-get update
+# Install gcc and Python 3 (via pip, which will bring along Python 3 as
+# a dependency).
+sudo apt-get install -y build-essential python3-pip
+# Install system packages required for the Python 3.9 conan package.
+# These would be installed as part of the conan package install, but
+# we're caching the conan directory via the `actions/cache` Github
+# action, so a fresh Github VM is left without these system packages.
+sudo apt-get install -y  --no-install-recommends libfontenc-dev libx11-xcb-dev libxaw7-dev \
+    libxcb-dri3-dev libxcb-icccm4-dev libxcb-image0-dev libxcb-keysyms1-dev libxcb-randr0-dev \
+    libxcb-render-util0-dev libxcb-shape0-dev libxcb-sync-dev libxcb-util-dev libxcb-xfixes0-dev \
+    libxcb-xinerama0-dev libxcb-xkb-dev libxcomposite-dev libxcursor-dev libxdamage-dev \
+    libxfixes-dev libxi-dev libxinerama-dev libxmu-dev libxmuu-dev libxpm-dev libxrandr-dev \
+    libxres-dev libxss-dev libxtst-dev libxv-dev libxvmc-dev libxxf86vm-dev
+
+# Install additional build tools.
+sudo pip3 install conan cmake==3.21 ninja
```

Comment:
[consider] Should we version pin `conan` and `ninja`? We had a lot of fun before with minor differences in package versions.

Reply:
Best to.
Validate wherever you can too.

CMake minimum version check

Conan can do 
e.g. https://github.com/conan-io/conan-center-index/blob/master/recipes/zlib/all/conanfile.py#L4